core/include/list.h

```diff
@@ -61,14 +64,37 @@ static inline void list_add(list_node_t *node, list_node_t *new_node) {
  *
  * @return  removed old list head, or NULL if empty
  */
-static inline list_node_t* list_remove_head(list_node_t *list) {
-    list_node_t* head = list->next;
+static inline list_node_t *list_remove_head(list_node_t *list)
+{
+    list_node_t *head = list->next;
+
     if (head) {
         list->next = head->next;
     }
     return head;
 }
 
+/**
+ * @brief Removes the node from the list
+ *
+ * @param[in] list  Pointer to the list itself, where list->next points
+ *                  to the root node
+ * @param[in] node  List node to remove from the list
+ *
+ * @return  removed node, or NULL if empty or not found
+ */
+static inline list_node_t *list_remove(list_node_t *list, list_node_t *node)
+{
+    while (list->next) {
+        if (list->next == node) {
+            list->next = node->next;
+            return node;
+        }
+        list = list->next;
+    }
+    return NULL;
```

Comment:
No import of `<stddef.h>` in this file (yet ;-)), so `NULL` is not defined (found by Murdock)